src/overdub.jl

```diff
@@ -144,7 +144,10 @@ function overdub_pass!(reflection::Reflection,
     #=== execute user-provided pass (is a no-op by default) ===#
 
     if !iskwfunc
-        code_info = passtype(context_type)(context_type, reflection)
+        # passtype is defined in the wrong world-age so we need to run it in the newest
+        # ideally we would add a edge from that function to the generated thunk, but
+        # alas that would require us, to trace all dynamic function calls. Here be dragons.
```

Comment:
Don't we already trace all function calls anyway?
It is used in StagedFunctions.jl

Reply:
It's a chicken and egg problem... StagedFunctions.jl uses Cassette to trace all functions executions https://github.com/NHDaly/StagedFunctions.jl/blob/43b8aa233c6fd765ca7c8b2f7cf912d796cbe3ac/src/StagedFunctions.jl#L76
so that edges are placed to dynamic functions correctly.

Reply:
To be clear, if we moved that logic into Julia, a la https://github.com/JuliaLang/julia/pull/33710, then it wouldn't be a chicken-and-egg problem; with the improved Generated functions from Base, Cassette would just work. :)

Until then, we'll need to do something here.